easybuild/easyblocks/generic/configuremake.py

```diff
@@ -83,11 +85,23 @@ def configure_step(self, cmd_prefix=''):
         if prefix_opt is None:
             prefix_opt = '--prefix='
 
-        cmd = "%(preconfigopts)s %(cmd_prefix)s./configure %(prefix_opt)s%(installdir)s %(configopts)s" % {
+        # Avoid using config.guess from the package as it is frequently out of date, use the version shipped with EB
+        build_type = self.cfg.get('build_type')
+        if build_type is None:
+            build_type, exit_code = run_cmd('config.guess')
+            if exit_code != 0:
+                raise EasyBuildError("config.guess shipped with EB could not determine build type: '%s'",
+                                     build_type)
+            build_type = build_type.strip()
+        build_type_option = '--build=' + build_type
+
+        cmd = ("%(preconfigopts)s %(cmd_prefix)s./configure %(prefix_opt)s%(installdir)s %(build_type_option)s "
```

Comment:
@ocaisa I like the `cmd = ' '.join([...])` option better, there's little point in using a dict here to format the command...